### DISCUSSION TITLE
local.properties is not being found

### DISCUSSION THREAD

**Discussion by belensaacastro** [codemagic-ci-cd/codemagic-docs#2047] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I'm dealing now with Android's new policy, and I need to update the minSdkVersion and targetSdkVersion.
As I'm using a flutter version greater than 2.8 I need to use a local.property file to set these constants, and after that set them on my build.gradle file its values.
Like this:
```
defaultConfig {

        minSdkVersion localProperties.getProperty('flutter.minSdkVersion').toInteger()
        targetSdkVersion localProperties.getProperty('flutter.targetSdkVersion').toInteger()
       
    }
```

I'm using Codemagic Workflow editor instead of the .yaml file, so my question for you is: How can I add these local.properties values to Codemagic?

I tried to add this according to the [post ](https://stackoverflow.com/questions/76791779/codemagic-warning-ignoring-settings-from-local-properties) in the Post-clone script:
```
#!/bin/sh
# set up local properties
echo "flutter.sdk=$HOME/programs/flutter" > "$CM_BUILD_DIR/android/local.properties"
echo "flutter.minSdkVersion=21" >> "$CM_BUILD_DIR/android/local.properties"
echo "flutter.targetSdkVersion=33" >> "$CM_BUILD_DIR/android/local.properties"
echo "flutter.compileSdkVersion=33" >> "$CM_BUILD_DIR/android/local.properties"
```

But I'm getting an error.

### Expected results

I'm expecting to build the app with the updated  minSdkVersion and  targetSdkVersion values.

### Actual results

Im getting this error from Codemagic tool:

/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_3_1_jhddagae: line 3: /Users/builder/clone/android/local.properties: No such file or directory
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_3_1_jhddagae: line 4: /Users/builder/clone/android/local.properties: No such file or directory
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_3_1_jhddagae: line 5: /Users/builder/clone/android/local.properties: No such file or directory
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_3_1_jhddagae: line 6: /Users/builder/clone/android/local.properties: No such file or directory

Build failed :|
Step 3 script `Post-clone script` exited with status code 1

### Build id (optional)

650379923886b4868392bb72

**Comment by dtrdic** [comment]
hey @belensaacastro, why are you using local.properties file to set the minSdkVersion and targetSdkVersion? can you try directly changing inside your android/app/build.gradle file?

**Comment by belensaacastro** [comment]
Hello David! That's a really good question...
According to this post
<https://www.woolha.com/tutorials/flutter-change-android-minsdkversion-targetsdkversion-compilesdkversion>
using
the local.properties was recommended instead of hardcoding the values when
you are using a Flutter version greather than 2.8...
I will try and I'll let you know.


By the way, if I would need to use some values from a file. What would be
the best approach to do that using the Workflow editor?
Why is not working the script I added on the Post cloning script?

Thank you for your early response!
Cheers,

Belen.

**Reply by dtrdic** [reply, marked as answer]
hey @belensaacastro, probably because `local.properties` file hasn't been created yet.
Try adding `echo "sdk.dir=$ANDROID_SDK_ROOT" > "$CM_BUILD_DIR/android/local.properties"` before setting up other properties, like this

```
echo "sdk.dir=$ANDROID_SDK_ROOT" > "$CM_BUILD_DIR/android/local.properties"

echo "flutter.sdk=$HOME/programs/flutter" > "$CM_BUILD_DIR/android/local.properties"
echo "flutter.minSdkVersion=21" >> "$CM_BUILD_DIR/android/local.properties"
echo "flutter.targetSdkVersion=33" >> "$CM_BUILD_DIR/android/local.properties"
echo "flutter.compileSdkVersion=33" >> "$CM_BUILD_DIR/android/local.properties"
```

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
